Modify S3247: Improve description and update example (#4019)
SonarSource/rspec: rules/S3247/csharp/rule.adoc

== Why is this an issue?

> In C#, the https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast#is-operator[`is` type testing operator] can be used to check if the run-time type of an object is compatible with a given type. If the check is successful, it means the code performed a cast. Performing another cast following the check result into a duplicate cast.

Reviewer: ```suggestion
In C#, the https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast#is-operator[`is` type testing operator] can be used to check if the run-time type of an object is compatible with a given type. If the object is not null, then the `is` operator performs a cast, and so performing another cast following the check result is redundant.
```
